tests/test_scene_optimizer.py

```diff
@@ -20,79 +20,75 @@
+def test_create_computation_graph_with_ground_truth(dataset_path):
+    print("running {}", dataset_path)
+    loader = OlssonLoader(str(dataset_path), image_extension="JPG")
+    assert len(loader)
+    """Will test Dask multi-processing capabilities and ability to serialize all objects."""
+    with hydra.initialize_config_module(config_module="gtsfm.configs"):
+
+        # Config is relative to the gtsfm module
+        cfg = hydra.compose(config_name="scene_optimizer_unit_test_config.yaml")
+        scene_optimizer: SceneOptimizer = instantiate(cfg.SceneOptimizer)
+
+        # Create dask client
+        cluster = LocalCluster(n_workers=1, threads_per_worker=4)
+        client = Client(cluster)
+
+        matching_regime = ImageMatchingRegime.EXHAUSTIVE
+        retriever = ExhaustiveRetriever()
+
+        pairs_graph = retriever.create_computation_graph(loader)
+
+        image_pair_indices = pairs_graph.compute()
+
+        (
+            delayed_keypoints,
+            delayed_putative_corr_idxs_dict,
+        ) = scene_optimizer.correspondence_generator.create_computation_graph(
+            delayed_images=loader.create_computation_graph_for_images(),
+            image_shapes=loader.get_image_shapes(),
+            image_pair_indices=image_pair_indices,
+        )
+
+        keypoints_list, putative_corr_idxs_dict = dask.compute(delayed_keypoints, delayed_putative_corr_idxs_dict)
+
+        delayed_sfm_result, delayed_io = scene_optimizer.create_computation_graph(
+            keypoints_list=keypoints_list,
+            putative_corr_idxs_dict=putative_corr_idxs_dict,
+            num_images=len(loader),
+            image_pair_indices=image_pair_indices,
+            image_graph=loader.create_computation_graph_for_images(),
+            all_intrinsics=loader.get_all_intrinsics(),
+            image_shapes=loader.get_image_shapes(),
+            absolute_pose_priors=loader.get_absolute_pose_priors(),
+            relative_pose_priors=loader.get_relative_pose_priors(image_pair_indices),
+            cameras_gt=loader.get_gt_cameras(),
+            gt_wTi_list=loader.get_gt_poses(),
+            matching_regime=ImageMatchingRegime(matching_regime),
+        )
+
+        sfm_result, *io = dask.compute(delayed_sfm_result, *delayed_io)
+
+        client.close()
+
+        assert isinstance(sfm_result, GtsfmData)
+
+        if dataset_path != TEST_DATA_NO_GT:
             # compare the camera poses
             computed_poses = sfm_result.get_camera_poses()
 
-            # get active cameras from largest connected component, may be <len(self.loader)
+            # get active cameras from largest connected component, may be <len(loader)
             connected_camera_idxs = sfm_result.get_valid_camera_indices()
-            expected_poses = [self.loader.get_camera_pose(i) for i in connected_camera_idxs]
+            expected_poses = [loader.get_camera_pose(i) for i in connected_camera_idxs]
```

Comment:
could we still check something on this result without ground truth? number of valid camera poses?

Reply:
@akshay-krishnan @ayushbaid Hey guys, did we address this?

Reply:
Yes, we now have `assert len(sfm_result.get_valid_camera_indices()) == len(loader)` when we do not have ground truth.